### PR TITLE
Expand websocket closure code handling

### PR DESCRIPTION
**Pull request by cosive-james-cooper**
This commit introduces a substantially expanded error handling block for when the websocket has closed unexpectedly, leading to a timeout error.  Specifically, now all 14 defined closure status codes are handled individually, rather than four of them.  Furthermore, the catch-all exception handler at the end has been modified to provide (slightly) more information that should hopefully be more useful to us in debugging errors on production boxes.

This also includes a bunch of small tidying measures taken in the chrome.py file, and adjusted the initial declaration location of a couple of variables, since it was possible those variables could be used without having first been declared.

Lastly, the requirements.txt files have been updated.  In requirements.txt the lxml requirement has been updated to a more recent version, while dev_requirements.txt has been changed to include requirements.txt in itself explicitly.

There will probably be several 'changes' in this PR that relate to whitespace only, so you might want to turn on the "no whitespace" option here.

### CODE REVIEW

**Comment by cosive-james-cooper** [comment]
Forgot to mention...  This should close Jira issue AUTO-515 if/when merged.

**Comment by cosive-james-cooper** [comment]
For what it is worth, we now seem to have been through several rounds of successful testing using the head commit on this branch.  I propose we consider again merging this back into master :)

File: setup.py [hunk 1]
```diff
@@ -7,27 +7,32 @@
...
     description='asyncio driver + HTTP server for Chrome devtools protocol',
     long_description=readme(),
+    # This project was originally forked from https://github.com/chuckus/chromewhip, but the version included here
+    # has been developed subsequently by Cosive, while the original version doesn't appear to have been maintained in
+    # some time.
     # The project's main homepage.
-    url='https://github.com/chuckus/chromewhip',
-    download_url='https://github.com/chuckus/chromewhip/archive/v0.3.4.tar.gz',
+    url='https://github.com/Cosive/chromewhip',
+    download_url='https://github.com/Cosive/chromewhip/archive/refs/tags/chrome-96.0.4664.110.tar.gz',
```
Comment (ahosie) [inline setup.py]: This download_url will be brittle, and we're not publishing this module to PyPI, so there's no value added here.

I strongly recommend leaving this (and all files sourced from a third-party repo) as-is - unmodified from the original source - unless a change is strictly required to minimise merge fights. Alternatively a (more) stable download URL could be set.

```suggestion
    download_url='https://codeload.github.com/Cosive/chromewhip/tar.gz/refs/heads/master',
```

File: setup.py [hunk 2]
```diff
@@ -7,27 +7,32 @@
...
     name='chromewhip',
 
-    version='0.3.4',
+    version='0.3.5',
 
     description='asyncio driver + HTTP server for Chrome devtools protocol',
     long_description=readme(),
+    # This project was originally forked from https://github.com/chuckus/chromewhip, but the version included here
+    # has been developed subsequently by Cosive, while the original version doesn't appear to have been maintained in
+    # some time.
```
Comment (ahosie) [inline setup.py]:
```suggestion

```

**Review by ahosie — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (cosive-james-cooper) [reply setup.py]: I'm moderately certain that the old download URL listed here was the one that was actually being used when downloading Chromewhip for use in other software (I can't remember precisely why I think that now, though).  Thus, it was quite important to update the URL from what it had been, i.e. an old one on the repo we don't control.

Your proposed new URL does seem likely to be the better choice, though.

**Pull request merged**